py4j-python/src/py4j/clientserver.py

```diff
@@ -203,6 +203,32 @@ class JavaClient(GatewayClient):
     ClientServerConnection per thread.
     """
 
+    class ThreadLocalConnectionCleaner(object):
```

Comment:
Should it be ThreadLocalConnectionFinalizer to better indicate its purpose (like the ThreadSafeFinalizer)?

Reply:
Also, does this class need to be inside JavaClient? This is a minor comment, but I try to only create internal structure (internal function and internal class) if they need to refer to the outer context.

Reply:
Let me take a closer look ..